pandas/tests/groupby/test_categorical.py

```diff
@@ -1380,3 +1380,16 @@ def test_groupby_agg_non_numeric():
 
     result = df.groupby([1, 2, 1]).nunique()
     tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize("func", ["first", "last"])
+def test_groupy_first_returns_categorical(func):
+    # GH 28641: groupby drops index, when grouping over categorical column with
+    # first/last
+    df = pd.DataFrame(
+        {"A": [1997], "B": pd.Series(["b"], dtype="category").cat.as_ordered()}
+    )
+    df_grouped = df.groupby("A")["B"]
+    result = getattr(df_grouped, func)()
+    expected = pd.Series(["b"], index=pd.Index([1997], name="A"), name="B")
+    tm.assert_series_equal(result, expected)
```

Comment:
im not sure i follow.  the test name says you want to get categorical, but the `expected` here is not categorical

Reply:
Sorry, the wording is obviously a bit confusing. The initial return value of the whole group by function was categorical and not a ``DataFrame`` as mentioned in [here](https://github.com/pandas-dev/pandas/issues/28641#issuecomment-549116890)

Should I change the name or write a comment?

Reply:
if you can come up with a concise+accurate name thatd be ideal

Reply:
Renamed the test. I think it reflects the previous issue a bit better now. Added a comment for good measure.